bundles/remote_services/discovery_common/src/discovery_activator.c

```diff
@@ -102,14 +102,8 @@ celix_status_t bundleActivator_start(void * userData, celix_bundle_context_t *co
 		return CELIX_ILLEGAL_STATE;
 	}
 
-	size_t len = 11 + strlen(OSGI_FRAMEWORK_OBJECTCLASS) + strlen(OSGI_RSA_ENDPOINT_FRAMEWORK_UUID) + strlen(uuid);
-	char *scope = malloc(len + 1);
-	if (!scope) {
-		return CELIX_ENOMEM;
-	}
-
-	sprintf(scope, "(&(%s=*)(%s=%s))", OSGI_FRAMEWORK_OBJECTCLASS, OSGI_RSA_ENDPOINT_FRAMEWORK_UUID, uuid);
-	scope[len] = 0;
+    char* scope = NULL;
+    asprintf(&scope, "(&(%s=*)(%s=%s))", OSGI_FRAMEWORK_OBJECTCLASS, OSGI_RSA_ENDPOINT_FRAMEWORK_UUID, uuid);
```

Comment:
Shall we check return value of `asprintf`? According to the manual page, its content is undefined in error.